posthog/warehouse/models/ssh_tunnel.py

```diff
@@ -1,8 +1,51 @@
 import dataclasses
 from io import StringIO
-from typing import Literal
+from typing import IO, Literal
 from sshtunnel import SSHTunnelForwarder
-from paramiko import RSAKey
+from paramiko import RSAKey, Ed25519Key, ECDSAKey, DSSKey, PKey
+from cryptography.hazmat.primitives import serialization as crypto_serialization
+from cryptography.hazmat.primitives.asymmetric import ed25519, dsa, rsa, ec
+
+
+# Taken from https://stackoverflow.com/questions/60660919/paramiko-ssh-client-is-unable-to-unpack-ed25519-key
+def from_private_key(file_obj: IO[str], passphrase: str | None = None) -> PKey:
```

Comment:
tests for different key types here? Non blocking if you want to merge this to resolve immediate issue